[ignition-transport-8] compile error protobuf

After protobuf update the package does not build anymore

Build log:
```
In file included from /build/ignition-transport-8/src/gz-transport-ignition-transport8_8.4.0/include/gz/transport/NodeShared.hh:41,
                 from /build/ignition-transport-8/src/gz-transport-ignition-transport8_8.4.0/include/gz/transport/Node.hh:42,
                 from /build/ignition-transport-8/src/gz-transport-ignition-transport8_8.4.0/src/CIface.cc:21:
/build/ignition-transport-8/src/gz-transport-ignition-transport8_8.4.0/include/gz/transport/RepHandler.hh:30:10: fatal error: google/protobuf/stubs/casts.h: No such file or directory
   30 | #include <google/protobuf/stubs/casts.h>
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
make[2]: *** [src/CMakeFiles/ignition-transport8.dir/build.make:90: src/CMakeFiles/ignition-transport8.dir/CIface.cc.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:346: src/CMakeFiles/ignition-transport8.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
[  8%] Built target doc
make: *** [Makefile:146: all] Error 2
==> ERROR: A failure occurred in build().
```
https://github.com/arch4edu/cactus/actions/runs/6219832833/job/16878662591

--- a/ignition-transport-8/.SRCINFO
+++ b/ignition-transport-8/.SRCINFO
@@ -15,6 +15,7 @@ pkgbase = ignition-transport-8
 	depends = ignition-msgs=5
 	depends = ignition-tools=1
 	depends = libutil-linux
+	depends = cppzmq
 	provides = ignition-transport=8
 	source = https://github.com/gazebosim/gz-transport/archive/ignition-transport8_8.4.0.tar.gz
 	source = ./protobuf-stubs-casts.patch

--- a/ignition-transport-8/.SRCINFO
+++ b/ignition-transport-8/.SRCINFO
@@ -1,7 +1,7 @@
 pkgbase = ignition-transport-8
 	pkgdesc = Transport library for component communication based on publication/subscription and service calls.
 	pkgver = 8.4.0
-	pkgrel = 2
+	pkgrel = 3
 	url = https://gazebosim.org/libs/transport
 	arch = i686
 	arch = x86_64
@@ -17,6 +17,8 @@ pkgbase = ignition-transport-8
 	depends = libutil-linux
 	provides = ignition-transport=8
 	source = https://github.com/gazebosim/gz-transport/archive/ignition-transport8_8.4.0.tar.gz
+	source = ./protobuf-stubs-casts.patch
 	sha256sums = 65472a874bbb0f6bb686b3c0cce4146fb650e750f8f795a159523ecbe731c1c7
+	sha256sums = c6490e7d8099dd11a00338f7440d7355ab28ca7f43cdbe3dea1db12bbd583d76
 
 pkgname = ignition-transport-8

--- a/ignition-transport-8/PKGBUILD
+++ b/ignition-transport-8/PKGBUILD
@@ -3,7 +3,7 @@
 
 pkgname=ignition-transport-8
 pkgver=8.4.0
-pkgrel=2
+pkgrel=3
 pkgdesc="Transport library for component communication based on publication/subscription and service calls."
 arch=('i686' 'x86_64')
 url="https://gazebosim.org/libs/transport"
@@ -13,10 +13,16 @@ depends=('protobuf' 'protobuf-c' 'zeromq' 'ignition-msgs=5' 'ignition-tools=1'
          'libutil-linux')
 makedepends=('ignition-cmake=2' 'util-linux')
 provides=('ignition-transport=8')
-source=("https://github.com/gazebosim/gz-transport/archive/ignition-transport8_${pkgver}.tar.gz")
-sha256sums=('65472a874bbb0f6bb686b3c0cce4146fb650e750f8f795a159523ecbe731c1c7')
+source=("https://github.com/gazebosim/gz-transport/archive/ignition-transport8_${pkgver}.tar.gz" "./protobuf-stubs-casts.patch")
+sha256sums=('65472a874bbb0f6bb686b3c0cce4146fb650e750f8f795a159523ecbe731c1c7'
+            'c6490e7d8099dd11a00338f7440d7355ab28ca7f43cdbe3dea1db12bbd583d76')
 
 _dir="gz-transport-ignition-transport8_${pkgver}"
+
+prepare() {
+    cd ${srcdir}/${_dir}
+    patch -Np1 -i ${srcdir}/protobuf-stubs-casts.patch
+}
 
 build() {
   cd "${srcdir}/${_dir}"

--- a/ignition-transport-8/PKGBUILD
+++ b/ignition-transport-8/PKGBUILD
@@ -10,7 +10,7 @@ url="https://gazebosim.org/libs/transport"
 license=('Apache')
 groups=('development')
 depends=('protobuf' 'protobuf-c' 'zeromq' 'ignition-msgs=5' 'ignition-tools=1'
-         'libutil-linux')
+         'libutil-linux' 'cppzmq')
 makedepends=('ignition-cmake=2' 'util-linux')
 provides=('ignition-transport=8')
 source=("https://github.com/gazebosim/gz-transport/archive/ignition-transport8_${pkgver}.tar.gz" "./protobuf-stubs-casts.patch")

--- a/ignition-transport-8/protobuf-stubs-casts.patch
+++ b/ignition-transport-8/protobuf-stubs-casts.patch
@@ -1,0 +1,48 @@
+diff --color -Naur gz-transport-ignition-transport8_8.4.0/include/gz/transport/RepHandler.hh gz-transport-ignition-transport8_8.4.0_new/include/gz/transport/RepHandler.hh
+--- gz-transport-ignition-transport8_8.4.0/include/gz/transport/RepHandler.hh   2022-11-18 00:34:01.000000000 +0100
++++ gz-transport-ignition-transport8_8.4.0_new/include/gz/transport/RepHandler.hh   2023-07-17 11:00:52.110403431 +0200
+@@ -26,7 +26,7 @@
+ #pragma warning(pop)
+ #endif
+
+-#if GOOGLE_PROTOBUF_VERSION > 2999999
++#if GOOGLE_PROTOBUF_VERSION > 2999999 && GOOGLE_PROTOBUF_VERSION < 4022000
+ #include <google/protobuf/stubs/casts.h>
+ #endif
+
+@@ -140,7 +140,11 @@
+           return false;
+         }
+
+-#if GOOGLE_PROTOBUF_VERSION > 2999999
++#if GOOGLE_PROTOBUF_VERSION >= 4022000
++        auto msgReq =
++          google::protobuf::internal::DownCast<const Req*>(&_msgReq);
++        auto msgRep = google::protobuf::internal::DownCast<Rep*>(&_msgRep);
++#elif GOOGLE_PROTOBUF_VERSION > 2999999
+         auto msgReq = google::protobuf::down_cast<const Req*>(&_msgReq);
+         auto msgRep = google::protobuf::down_cast<Rep*>(&_msgRep);
+ #else
+diff --color -Naur gz-transport-ignition-transport8_8.4.0/include/gz/transport/SubscriptionHandler.hh gz-transport-ignition-transport8_8.4.0_new/include/gz/transport/SubscriptionHandler.hh
+--- gz-transport-ignition-transport8_8.4.0/include/gz/transport/SubscriptionHandler.hh  2022-11-18 00:34:01.000000000 +0100
++++ gz-transport-ignition-transport8_8.4.0_new/include/gz/transport/SubscriptionHandler.hh  2023-07-17 11:01:36.354294479 +0200
+@@ -28,7 +28,7 @@
+
+ #include <google/protobuf/stubs/common.h>
+
+-#if GOOGLE_PROTOBUF_VERSION >= 3000000
++#if GOOGLE_PROTOBUF_VERSION >= 3000000 && GOOGLE_PROTOBUF_VERSION < 4022000
+ #include <google/protobuf/stubs/casts.h>
+ #endif
+
+@@ -211,7 +211,9 @@
+         if (!this->UpdateThrottling())
+           return true;
+
+-#if GOOGLE_PROTOBUF_VERSION >= 3000000
++#if GOOGLE_PROTOBUF_VERSION >= 4022000
++        auto msgPtr = google::protobuf::internal::DownCast<const T*>(&_msg);
++#elif GOOGLE_PROTOBUF_VERSION >= 3000000
+         auto msgPtr = google::protobuf::down_cast<const T*>(&_msg);
+ #else
+         auto msgPtr = google::protobuf::internal::down_cast<const T*>(&_msg);

